### DISCUSSION TITLE
Create a thermostat_over_pilotwire

### DISCUSSION THREAD

**Discussion by celestianx** [jmcollin78/versatile_thermostat#268] [Ideas]
While I reckon this might be considered as a redundant feature of the "themostat_over_switch", I own Legand Cable outlet which can operate both as switch or pilotwire mode. The switch mode makes the outlet contactor click which I find annoying.
I have already developed a climate entity based on the generic_thermostat overwriting to send the pilotwire command.
I would love to see that for versatile_thermostat and actually I have already started to adapt the code to add a new type of thermostat and need to test it but before proceeding any further, wanted to discuss it to know if you would support this new addition once I have developed it and tested it or if you have any additional valuable thoughts

**Comment by jmcollin78** [comment]
Hello, pilot wire are already taken into account in VTherm via a template switch in between.

If you think your change can remove this intermediary template switch, it will be welcomed. But I think this is not so trivial due to many equipments that are not working in a generic way.

Please give me more information before continuing. If it is too much specific i will not want to take it.

**Reply by celestianx** [reply]
I didn’t know such templates existed. Knowing this, I agree with you that only if they works all the same, it would not be a good idea to include the logic in vtherm

As for Legrand it is a simple select mechanism from z2m between comfort, eco, comfort-1, comfort-2, frozen and off. I will let you judge if you have seen other implementations that differ but I doubt that in HA at least, it would be any different as from a user experience perspective this is the most straightforward to select between different modes

**Reply by jmcollin78** [reply]
You have an example of such a template in the documentation here: https://github.com/jmcollin78/versatile_thermostat/tree/main#using-a-heatzy
